nexus/types/src/inventory.rs

```diff
@@ -124,7 +123,7 @@ pub struct Collection {
     /// cluster as returned from each available keeper via `clickhouse-admin` in
     /// the `ClickhouseKeeper` zone
```

Comment:
Everything else here is a map -- might be worth adding a short explanation for why this is keyed by the membership itself rather than something else (like the zone ID that it previously was).

Reply:
We don't actually know the zone id when we look up the address via DNS, so using that as a key was problematic. However, we could destructure the returned value and map via KeeperId. But then we end up putting it back together again so we can store it as a value in the blueprint. I think this is pretty tedious and doesn't buy us anything. If there was ever a duplicate for some reason with slightly different values we'd end up overwriting early versions with later versions which could hinder debugging. I guess this latter comment leans somewhat towards this being a `Vec` instead then. 

I could change how we access the `clickhouse-admin` service rather than looking it up in DNS, and do a DB query instead and then map by zone. But then we end up simply discarding the Zone ID downstream since there's no use for it.

I'm happy to had a small comment, but I'm really unsure what to put there. I can put what I said here but that seems somewhat unsatisfying and I'm unsure of the value.

Reply:
I think what you said here is very valuable! Either the full comment, or a shortened form with a link to this comment would be great.

Reply:
Could the clickhouse admin server return the zone ID from the endpoint we hit? For `cockroach-admin` we have sled-agent pass in the zone ID, so it can return it from (one of) the endpoints for similar reasons.

Reply:
Is it actually useful to do that though? The mapping from keeper id to zone id already exists in the blueprints.

Reply:
I added a long comment detailing how this is used and how the reconfigurator should prevent any duplicates. We could cache and use the zone IDs for keys, but I'm not sure that provides much in terms of safety over what is already there.